src/runtime/cuda.cpp

```diff
@@ -552,19 +605,48 @@ WEAK int halide_cuda_device_free(void *user_context, halide_buffer_t* buf) {
         << ", buf: " << buf << ")\n";
 
     Context ctx(user_context);
-    if (ctx.error != CUDA_SUCCESS)
+    if (ctx.error != CUDA_SUCCESS) {
         return ctx.error;
+    }
 
     #ifdef DEBUG_RUNTIME
     uint64_t t_before = halide_current_time_ns(user_context);
     #endif
 
     halide_assert(user_context, validate_device_pointer(user_context, buf));
 
-    debug(user_context) <<  "    cuMemFree " << (void *)(dev_ptr) << "\n";
-    CUresult err = cuMemFree(dev_ptr);
-    // If cuMemFree fails, it isn't likely to succeed later, so just drop
-    // the reference.
+    CUresult err = CUDA_SUCCESS;
```

Comment:
If possible, I think this should be in a helper function wrapper around cuMemAlloc/cuMemFree, even if this helper function is not something that is overrideable a custom runtime (re: our gitter conversation about this).